src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -144,6 +146,20 @@ def image_refs(
             for registry in registries_options
         )
 
+    def get_context_root(self, default_context_root: str) -> str:
+        context_root = self.context_root.value or default_context_root
+        if context_root.startswith("BUILD:"):
```

Comment:
I'm not sure about this. The name `BUILD` is configurable, so the BUILD file might not actually be called BUILD and this might lead to confusion. What if we look for a `./` prefix?

Reply:
I'm looking for a way to say that the path should be relative to the BUILD file, rather than the project root (which is the default) so src/foo would be from the build root. So it feels counter intuitive then to have ./bar to not be relative to the build root..

Maybe leave out BUILD and have only :, that mimics the syntax for sibling target addresses

:./bar
:bar  # hmm, that looks too much like an address, but if we require the ./ then I think it could work.

Reply:
Or, what if we use lowercase so it refers to the concept, rather than the filename?

build:./bar

Reply:
Lower-case `build` seems like a gratuitious new concept, TBH

Reply:
To me `src/foo` is relative to the build root (it's shorthand for `//src/foo`) but `./src/foo` is relative to the BUILD file seems reasonably intuitive?

Reply:
Can bikeshed it...

Reply:
> Can bikeshed it...

No need.

> To me src/foo is relative to the build root (it's shorthand for //src/foo) but ./src/foo is relative to the BUILD file seems reasonably intuitive?

I'm absolutely on board with this approach. (and given the implicit `//` explanation makes this scenario clear to me as well).

Reply:
This is already the approach we take with file addresses in the `dependencies` field.